Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -0,0 +1,544 @@
+
+        switch (type) {
+        case Type::Byte:
+        case Type::Undefined:
+            return read_every_values.template operator()<u8>();
+        case Type::ASCII:
+        case Type::UTF8: {
+            Vector<Value, 1> result;
+            auto string_data = TRY(ByteBuffer::create_uninitialized(count));
+            TRY(m_stream->read_until_filled(string_data));
+            result.empend(TRY(String::from_utf8(StringView { string_data.bytes() })));
+            return result;
+        }
+        case Type::Short:
+            return read_every_values.template operator()<u16>();
+        case Type::Long:
+            return read_every_values.template operator()<u32>();
+        case Type::Rational:
+            return read_every_values.template operator()<Rational<u32>>();
+        case Type::SLong:
+            return read_every_values.template operator()<i32>();
+            ;
+        case Type::SRational:
+            return read_every_values.template operator()<Rational<i32>>();
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    }
+
+    ErrorOr<void> read_tag()
+    {
+        auto const tag = TRY(read_value<u16>());
+        auto const type = TRY(read_type());
+        auto const count = TRY(read_value<u32>());
+
+        Checked<u32> checked_size = size_of_type(type);
+        checked_size *= count;
+
+        if (checked_size.has_overflow())
+            return Error::from_string_literal("TIFFImageDecoderPlugin: Invalid tag with too large data");
+
+        auto const tiff_value = TRY(([=, this]() -> ErrorOr<Vector<Value>> {
+            if (checked_size.value() <= 4) {
+                auto value = TRY(read_tiff_value(type, count, TRY(m_stream->tell())));
+                TRY(m_stream->discard(4));
+                return value;
+            }
+            auto const offset = TRY(read_value<u32>());
+            return read_tiff_value(type, count, offset);
+        }()));
+
+        if constexpr (TIFF_DEBUG) {
+            if (tiff_value.size() == 1) {
+                tiff_value[0].visit(
+                    [&](auto const& value) {
+                        dbgln("Read tag({}), type({}): {}", tag, to_underlying(type), value);
+                    });
+            } else {
+                dbg("Read tag({}), type({}): [", tag, to_underlying(type));
+                for (u32 i = 0; i < tiff_value.size(); ++i) {
+                    tiff_value[i].visit(
+                        [&](auto const& value) {
+                            dbg("{}", value);
+                        });
+                    if (i != tiff_value.size() - 1)
+                        dbg(", ");
+                }
+                dbgln("]");
+            }
+        }
+
+        TRY(handle_tag(tag, type, count, tiff_value));
+
+        return {};
+    }
+
+    ErrorOr<void> handle_tag(u16 tag, Type type, u32 count, Vector<Value> const& value)
+    {
+        // FIXME: Make that easy to extend
+        switch (tag) {
+        case 256:
```

Comment:
can we get named constants for these?

Reply:
I know that these magic numbers aren't great, but that's something that we should address by generating this code. Naming these will only put distance between the number and its usage IMO, a comment already describes what the number is.